### DISCUSSION TITLE
Not able to connect to node after renaming the node hostname

### DISCUSSION THREAD

**Discussion by SaiKiranAre** [gravitational/teleport#29294] [Q&A] [answered]
Hi There,

I am using teleport version ```7.3.26```. And able to add a SSH Node to cluster and its working fine. Recenly, we changed the hostname of the node. Afterwards, not able to connect the VSI using ```tsh ssh``` command.

The following error is coming:

```
ERROR: failed connecting to node xxxxxxxx. Teleport proxy failed to connect to "node" agent "ip:3022" over reverse tunnel:

  no tunnel connection found: no node reverse tunnel for xxxxxxxxx found

This usually means that the agent is offline or has disconnected. Check the
agent logs and, if the issue persists, try restarting it or re-registering it
with the cluster
```

Removed /var/lib/telport file , uninstalled teleport and re-registered .. Still the same problem is coming. And also removed the node using ```tctl rm node/id``` and re-registered still the same issue.

Observation: After re-register it is still showing ... up time as 5 days .. Not sure why it is like that.

Thanks a lot

**Comment by webvictim** [comment, marked as answer]
Hey - I'm afraid we aren't able to support Teleport v7 as it's several years old. If you upgrade your cluster to Teleport v13 and can still reproduce this bug, we can take a look at it. Thanks.

**Comment by SaiKiranAre** [comment]
We are in the process of upgrading to v13, It may take couple of weeks. In the mean time, it will be great if you suggest if there is any way to completely clear the node details from the server

**Reply by webvictim** [reply]
Shutting down Teleport completely, deleting `/var/lib/teleport` from the node and restarting Teleport should re-register the node with a brand new ID.

If this isn't happening, it's probably because there is still another `teleport` process running on the server which is heartbeating the old hostname. Make sure all processes have shut down - check `ps -ef | grep teleport | grep -v grep`

**Reply by SaiKiranAre** [reply]
Thanks a lot, will check that

**Reply by SaiKiranAre** [reply]
I tried and saw that, it given new id when I am listing ```tctl nodes ls```. But the up time is showing from last 5 days. Not sure from where it is picking the up time. Ideally it should show from last few mins.

**Reply by webvictim** [reply]
Maybe you can get the new ID from `tctl nodes ls` and share the output of `tctl get node/<uuid>`? The `expires:` time there in that output should be within the last minute or so if your node is actively sending heartbeats.

**Reply by SaiKiranAre** [reply]
Thanks a lot, It found out that after renaming the hostname the ufw and iptables got some issues. Removed the ufw chains in iptables and re enabled ufw solved the problem.

I got confused with up time, now I understood, this is OS uptime not the bastion connection up time.

Thanks a lot

**Comment by SaiKiranAre** [comment]
One more query, if SSH Node uses teleport 7.3.26 and teleport server is 13, then the tsh ssh connections works? This is for migration to 13.  Note: The SSH Node update should happen through tsh ssh connection only

Which is the better approach

1) Update SSH Nodes to V13 first and later teleport server to 13
2) Update teleport server to v13 first and later ssh node to 13

**Reply by webvictim** [reply]
A Teleport node on v7 will not connect to an auth server running v13. You should upgrade the auth server first and make sure to have a backup connection open to the Teleport node so you can upgrade the Teleport binary there afterwards.
